File: speed/mat.m
# mesuring performance of Octave

n = 500;

a = linspace(0,n*n-1,n*n);
a = rem(a, n+1) + 1;
a = reshape(a,n,n);

b = linspace(0,n*n-1,n*n);
b = rem(b, n-1) + 1;
b = reshape(b,n,n);

printf ("executing product of %ix%i Matrix...\n",n,n);
printf ("c=a*b\n");

[t1, u1, s1] = cputime ();
c = a*b;
[t2, u2, s2] = cputime ();

printf (" Time: %f\n", u2 - u1);
# Time: 7.280000
# GNU Octave, version 2.0.16 (sparc-sun-solaris2.7).
exit

File: speed/mat.rb
require 'mybench'

n = 500

a = NArray.float(n,n).indgen!
a = a % (n+1) + 1
a = NMatrix.ref(a)#.transpose

b = NArray.float(n,n).indgen!
b = b % (n-1) + 1
b = NMatrix.ref(b)#.transpose

c = 0

puts 'a='
p a
puts 'b='
p b

printf "executing %ix%i Matrix product...\n",n,n
puts 'c=a*b'

bench_time(1) { c=a*b }

# time: 4.21 sec

puts 'c='
p c
